src/components/client/search-modal.tsx

```diff
@@ -0,0 +1,277 @@
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  }, []);
+
+  React.useEffect(() => {
+    if (searchQuery !== prevQueryRef.current) {
+      if (
+        (searchQuery?.query?.author === "" ||
+          searchQuery?.query?.author === undefined) &&
+        (searchQuery?.query?.keyword === "" ||
+          searchQuery?.query?.keyword === undefined)
+      ) {
+        return;
+      }
+      startTransition(() => {
+        getData();
+      });
+      return () => {
+        debouncedSearch.cancel();
+        debouncedSearchAuthor.cancel();
+      };
+    }
+
+    prevQueryRef.current = searchQuery;
+  }, [
+    searchQuery,
+    startTransition,
+    getData,
+    debouncedSearch,
+    debouncedSearchAuthor,
+  ]);
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      searchFeedInputRef.current?.focus();
+    }, 100);
+
+    if (!open) {
+      setLimit(4);
+    }
+
+    return () => clearTimeout(timer);
+  }, [open]);
+
+  return (
+    <Dialog.Root open={open} onOpenChange={setOpen}>
+      <Dialog.Trigger asChild>
+        <div className="relative">
+          <input
+            name="searchbox"
+            className="border-[1px] rounded-md border-[#ccc] w-[200px] md:w-[274px] py-2 pl-9"
+            placeholder="search [cmd/ctrl+k]"
+          />
+          <Image
+            className="absolute top-1/2 left-0 -translate-y-1/2 ml-2 pointer-events-none"
+            src="/icons/search_icon.svg"
+            alt="search_icon"
+            width={24}
+            height={24}
+          />
+        </div>
+      </Dialog.Trigger>
+      <Dialog.Portal>
+        <Dialog.Overlay className="bg-slate-400 opacity-50 data-[state=open]:animate-overlayShow fixed inset-0" />
+        <Dialog.Content className="data-[state=open]:animate-contentShow overflow-y-auto fixed top-[50%] left-[50%] max-h-[65vh] w-[90vw] max-w-[450px] translate-x-[-50%] translate-y-[-50%] rounded-[6px] bg-white p-[25px] shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] focus:outline-none">
+          <Dialog.Title className="m-0 text-[16px] font-medium flex items-center gap-x-4">
+            <span>Filter results by: </span>
+            <div className="flex items-center gap-x-2">
+              <button
+                className={`border rounded py-1 px-2 text-xs hover:bg-slate-100 ${
+                  filter.bitcoinDev && "bg-slate-100"
+                }`}
+                onClick={() => {
+                  dispatch({ type: "bitcoinDev" });
+                  setSearchQueryPath("bitcoin-dev");
+                }}
+              >
+                bitcoin-dev
+              </button>
+              <button
+                className={`border rounded py-1 px-2 text-xs hover:bg-slate-100 ${
+                  filter.lightningDev && "bg-slate-100"
+                }`}
+                onClick={() => {
+                  dispatch({ type: "lightningDev" });
```

Comment:
we have a type `LIGHTNINGDEV` that resolves to "lightning-dev" to keep consistent naming for example between the action dispatch and searchQueryPath